docs/markdown/Using Pants/generating-version-tags.md

```diff
@@ -0,0 +1,64 @@
+---
+title: "Generating version tags from Git"
+slug: "generating-version-tags"
+hidden: false
+---
+
+Pants can generate a version string based on Git state that you can use during builds.
+
+The `vcs_version` target
+------------------------
+
+To utilize this functionality, you first create a `vcs_version` target in some BUILD file:
+
+```python src/BUILD
+vcs_version(
+    name="version",
+    generate_to="src/version.py",
+    template='version = {version}',
+)
+```
+
+When you test, run or package any code that depends (directly or indirectly) on this target, 
+Pants will compute a version string based on the current Git state, and then generate content 
+using the specified `template` (with `{version}` serving as a placeholder for the version string)
+at the path specified by `generate_to`.
+
+Note that, similarly to how Pants handles other code generation tools (such as [Protobuf](doc:protobuf-python)), 
+this content is not written out as a file into your source tree. Instead, the content is materialized
+on the fly, as needed.  If you want to inspect the generated contents manually, you can use the
+`export-codegen` goal:
+
+```shell
+./pants export-codegen src:version
+```
+
+Using the generated version
+---------------------------
+
+Code that depends on the `vcs_version` target can import and use the generated file:
+
```

Comment:
This sentence makes me think that this dependency can not be inferred. It is cleared up in a later paragraph however, if I read that far...

Reply:
It can only be inferred in a rather specific case, so I'd rather not get into that yet.  And "depends on" is fairly neutral as to whether the dep was explicit or inferred, no?